ocaml/testsuite/tests/typing-layouts/basics_alpha.ml

```diff
@@ -182,11 +182,12 @@ module F2 (X : sig val f : void_record -> unit end) = struct
   let g z = X.f { vr_void = z; vr_int = 42 }
 end;;
 [%%expect{|
-Line 2, characters 8-44:
+Line 2, characters 16-44:
 2 |   let g z = X.f { vr_void = z; vr_int = 42 }
-            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: Non-value layout void detected in [Typeopt.layout] as sort for type
-       t_void. Please report this error to the Jane Street compilers team.
+                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
The error message changed because now we compute the return (`Lambda.`)layout of a function before the args' layouts. And, computing the value kind of a record involves computing the value kind of all of its fields, which makes the safety check raise. The location available to this check is the location of the entire function body.

I'm inclined to resolve this: you can get similarly-confusing errors in the current world anyway:

```ocaml
# type ('a : void) t = { field : 'a; b : int };;
type ('a : void) t = { field : 'a; b : int; }
# let f (x : _ t) = ();;
        ^^^^^^^^^^^^^^
Error: Non-value detected in [value_kind].
       Please report this error to the Jane Street compilers team.
       'a has layout void, which is not a sublayout of value.
```